allenact/algorithms/onpolicy_sync/runner.py

```diff
@@ -810,30 +853,42 @@ def process_eval_package(
 
         num_tasks = pkg.num_non_empty_metrics_dicts_added
         metric_means = pkg.metrics_tracker.means()
+        callback_metric_means = dict()
+        tasks_callback_data = pkg.task_callback_data
 
         mode = pkg.mode
 
         if log_writer is not None:
             log_writer.add_scalar(
                 f"{mode}-misc/num_tasks_evaled", num_tasks, training_steps
             )
+            callback_metric_means[f"{mode}-misc/num_tasks_evaled"] = num_tasks
 
         message = [f"{mode} {training_steps} steps:"]
         for k in sorted(metric_means.keys()):
             if log_writer is not None:
                 log_writer.add_scalar(
                     f"{mode}-metrics/{k}", metric_means[k], training_steps
                 )
+                callback_metric_means[f"{mode}-metrics/{k}"] = metric_means[k]
             message.append(f"{k} {metric_means[k]}")
 
+        results = copy.deepcopy(metric_means)
+        results.update({"training_steps": training_steps, "tasks": task_outputs})
         if all_results is not None:
-            results = copy.deepcopy(metric_means)
-            results.update({"training_steps": training_steps, "tasks": task_outputs})
             all_results.append(results)
 
         message.append(f"tasks {num_tasks} checkpoint {checkpoint_file_name}")
         get_logger().info(" ".join(message))
 
+        for callback in self.callbacks:
+            callback.on_valid_log(
```

Comment:
You're right that this `process_eval_package` function is only being called to process validation packages. Can we rename `process_eval_package` to `process_val_package` (or `process_validation_package`) to make this more clear?

Reply:
We should also throw in a check that `mode == "valid"` somewhere in this function.